Source/CarthageKit/Project.swift

```diff
@@ -591,6 +591,16 @@ public final class Project { // swiftlint:disable:this type_body_length
 			.flatMap(.concat, unarchive(archive:))
 			.flatMap(.concat) { directoryURL -> SignalProducer<URL, CarthageError> in
 				return frameworksInDirectory(directoryURL)
+					.collect()
+					// Check if the Archive contains multiple framewroks with the same `lastPathComponent`, i.e. their name.
+					// This check is needed because they will be copied over to the same destination location.
+					.flatMap(.merge) { frameworksUrls -> SignalProducer<URL, CarthageError> in
+						let setOfUniqueFrameworkNames = Set(frameworksUrls.map { $0.lastPathComponent })
+						guard setOfUniqueFrameworkNames.count == frameworksUrls.count else {
+							return SignalProducer(error: CarthageError.invalidArchive(description: "found multiple frameworks with the same name"))
+						}
+						return SignalProducer(frameworksUrls)
+					}
```

Comment:
This doesn't look like it correctly handles multiple platforms. e.g. ReactiveCocoa supports all 4 platforms, so it can contain 4 `ReactiveCocoa.framework`s.

Reply:
@mdiep This is why there are code reviews 😅 

It now builds the full destination path. I force pushed on amend so this comment is now misplaced.